homeassistant/components/battery.py

```diff
@@ -0,0 +1,180 @@
+"""
+components.battery
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+battery:
+  iphonebart:
+    entity_id: device_tracker.iphonebart
+    type: attribute
+    attribute: battery
+  vera:
+    entity_id: vera.deviceid
+    type: attribute
+    attribute: battery_level
+  mysensor:
+    entity_id: sensor.mysensor
+    type: state
+"""
+
+import logging
+from homeassistant.helpers.event import track_state_change
+from homeassistant.helpers.entity import Entity
+from homeassistant.const import ATTR_BATTERY_LEVEL
+
+DEPENDENCIES = ['device_tracker']
+
+# domain for the component
+DOMAIN = 'battery'
+
+# entity attributes
+ATTR_STATUS = 'status'
+ATTR_BATTERY_TYPE = 'battery type'
+ATTR_ATTRIBUTE = 'attribute'
+
+# default type
+DEFAULT_TYPE = 'attribute'
+
+# default attribute
+DEFAULT_ATTRIBUTE = ATTR_BATTERY_LEVEL
+
+# default unit_of_measurement
+DEFAULT_UNIT = '%'
+
+# default charged_level
+DEFAULT_CHARGED = 100
+
+# Shortcut for the logger
+_LOGGER = logging.getLogger(__name__)
+
+
+def setup(hass, config):
+    """ get the battery details from configuration.yaml"""
+
+    batteries = []
+    if config.get(DOMAIN) is None:
+        return False
+
+    for bat, bat_config in config[DOMAIN].items():
+
+        if not isinstance(bat_config, dict):
+            _LOGGER.error("Missing configuration data for battery %s", bat)
+            continue
+
+        battery_type = bat_config.get('type', DEFAULT_TYPE)
+        attribute = bat_config.get('attribute', DEFAULT_ATTRIBUTE)
+        device_entity_id = bat_config.get('entity_id')
+        unit_of_measurement = bat_config.get('unit_of_measurement',
+                                             DEFAULT_UNIT)
+        charged_level = bat_config.get('charged_level', DEFAULT_CHARGED)
+
+        entity_id = DOMAIN + '.' + bat
+
+        battery = Battery(hass, battery_type, attribute, device_entity_id,
+                          unit_of_measurement, charged_level)
+        battery.entity_id = entity_id
```

Comment:
I think you should implement the name property in the entity class instead and return what you set here. For example:
```
battery._name = bat
...
@property
def name(self):
    """The name of this entity."""
    return self._name
```

The entity_id will be generated automatically using the name property when the entity is added in hass.